### PR TITLE
[feat] Add classification fine-tuning utilities

### PR DESCRIPTION
**Pull request by apsdehal**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #10
* #9
* __->__ #8

- The PR aims at ending starter classification utils to flava examples.

As of now the PR adds following things:
- Finetuning trainer
- Classification FLAVA
- TorchVisionDataModule for easy composability of datasets from
torchvision
- Some changes to MLP module for more generalization
- Some improvements/bug fixes to original FLAVA code
- Splits the datamodules to better service their individual concerns.

TODOs:
- Add support for rest of the datasets. This involves levaraging the
existing datamodules that we created in this PR along with support for
seamlessly plugging different dataset
- Add command line overriding on top
- Add support for retrieval, zero-shot and other downstream tasks in an
easily accessible form
- Expose more things from the model other than just the loss

Test Plan:

The code is not in 100% working stage. I have tested only the changes in
my PR. I expect everything to be stable by the end of the stack.

Differential Revision: [D35361821](https://our.internmc.facebook.com/intern/diff/D35361821)

### CODE REVIEW

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35271487).

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35271487).

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35361821).

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35361821).

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35361821).

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35361821).

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35361821).

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35361821).

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35361821).
